mixes/containers/mix.js

```diff
@@ -58,7 +59,20 @@ module.exports = connect(
   (state, ownProps) => {
     const props = getMixProps(state)
     const currentMixId = ownProps.params.mixId
-    return { ...props, mixId: currentMixId, mix: props.mixes[currentMixId] }
+    const { router, route } = ownProps
+    const mix = props.mixes[currentMixId]
+
+    if (mix) {
+      router.setRouteLeaveHook(
+        route,
+        // TODO: do i need to back out all changes if confirm? how to do that cleanly - LOAD_MIX? ROLLBACK_MIX?
```

Comment:
thoughts? seems like if we click 'ok' from this dialog, is the user expectation that the mix is rolled back (ie unsaved changes should be unloaded from the store)?